### PR TITLE
AT1-170: Integrate KATPoint (Ra-Dec to Az-El Conversion) into Dish Leaf Node

### PR DESCRIPTION
**Pull request by apurvapatkar**

### CODE REVIEW

File: tmcprototype/DishLeafNode/DishLeafNode/DishLeafNode.py [hunk 1]
```diff
@@ -210,7 +213,81 @@ def commandCallback(self, event):
             print(CONST.ERR_EXCEPT_CMD_CB, except_occurred)
             self._read_activity_message = CONST.ERR_EXCEPT_CMD_CB + str(except_occurred)
             self.dev_logging(CONST.ERR_EXCEPT_CMD_CB, int(tango.LogLevel.LOG_ERROR))
-    # PROTECTED REGION END #    //  DishLeafNode.class_variable
+
+
+    def convert_radec_to_azel(self, data):
+        try:
+            print ("data is:", data)
```
Comment (jyotinpo) [inline tmcprototype/DishLeafNode/DishLeafNode/DishLeafNode.py]: Remove unnecessary print statements.

File: tmcprototype/DishLeafNode/DishLeafNode/DishLeafNode.py [hunk 2]
```diff
@@ -210,7 +213,81 @@ def commandCallback(self, event):
...
+            print(type(data))
+            # Setting Observer Position as Pune
+            a = katpoint.Antenna(name='d1', latitude='18:31:48:00', longitude='73:50:23.99', altitude=570)
```
Comment (jyotinpo) [inline tmcprototype/DishLeafNode/DishLeafNode/DishLeafNode.py]: Use of variable names that makes sense.

File: tmcprototype/SubarrayNode/SubarrayNode/SubarrayNode.py [hunk 3]
```diff
@@ -418,14 +418,17 @@ def Configure(self, argin):
         :return: None
         """
         try:
-            for i in range(0, len(argin)):
-                if type(float(argin[i])) == float:
-                    pass
+            # Validation to check float argins
```
Comment (jyotinpo) [inline tmcprototype/SubarrayNode/SubarrayNode/SubarrayNode.py]: Remove commented code if not required.

File: tmcprototype/DishMaster/DishMaster/DishMaster.py [hunk 4]
```diff
@@ -59,6 +60,10 @@ def point(self):
         if((self._achieved_pointing[1] != self._desired_pointing[1]) |
            (self._achieved_pointing[2] != self._desired_pointing[2])):
             try:
+                self._azimuth_difference = self._desired_pointing[1] - self._achieved_pointing[1]
+                print("azimuth difference is: ", self._azimuth_difference)
+                self._elevation_difference = self._desired_pointing[2] - self._achieved_pointing[2]
+                print("elevation difference is: ", (self._elevation_difference))
                 self.change_azimuth_thread = threading.Thread(None, self.azimuth, 'DishMaster')
                 self.change_elevation_thread = threading.Thread(None, self.elevation, 'DishMaster')
                 self.change_azimuth_thread.start()
```
Comment (jyotinpo) [inline tmcprototype/DishMaster/DishMaster/DishMaster.py]: Is it also not required to start self.change_elevation_thread?

File: tmcprototype/DishLeafNode/DishLeafNode/DishLeafNode.py (hunk 1 above)
Comment (apurvapatkar) [reply tmcprototype/DishLeafNode/DishLeafNode/DishLeafNode.py]: Done.

File: tmcprototype/DishLeafNode/DishLeafNode/DishLeafNode.py (hunk 2 above)
Comment (apurvapatkar) [reply tmcprototype/DishLeafNode/DishLeafNode/DishLeafNode.py]: Done.

File: tmcprototype/DishMaster/DishMaster/DishMaster.py (hunk 4 above)
Comment (apurvapatkar) [reply tmcprototype/DishMaster/DishMaster/DishMaster.py]: Yes. It is required and self.change_elevation_thread start statement is already there at line 70 in DishMaster.py

File: tmcprototype/SubarrayNode/SubarrayNode/SubarrayNode.py (hunk 3 above)
Comment (apurvapatkar) [reply tmcprototype/SubarrayNode/SubarrayNode/SubarrayNode.py]: Done.

File: tmcprototype/DishLeafNode/DishLeafNode/DishLeafNode.py (hunk 1 above)
Comment (apurvapatkar) [reply tmcprototype/DishLeafNode/DishLeafNode/DishLeafNode.py]: Commit id is: [01a44358029a01fcc943c419fc4d7606eab518be](https://github.com/ska-telescope/tmc-prototype/commit/01a44358029a01fcc943c419fc4d7606eab518be)

File: tmcprototype/DishLeafNode/DishLeafNode/DishLeafNode.py (hunk 2 above)
Comment (apurvapatkar) [reply tmcprototype/DishLeafNode/DishLeafNode/DishLeafNode.py]: Commit id is: [01a44358029a01fcc943c419fc4d7606eab518be](https://github.com/ska-telescope/tmc-prototype/commit/01a44358029a01fcc943c419fc4d7606eab518be)

File: tmcprototype/SubarrayNode/SubarrayNode/SubarrayNode.py (hunk 3 above)
Comment (apurvapatkar) [reply tmcprototype/SubarrayNode/SubarrayNode/SubarrayNode.py]: Commit id is: [01a44358029a01fcc943c419fc4d7606eab518be](https://github.com/ska-telescope/tmc-prototype/commit/01a44358029a01fcc943c419fc4d7606eab518be)

File: tmcprototype/DishLeafNode/DishLeafNode/DishLeafNode.py [hunk 5]
```diff
@@ -416,15 +461,28 @@ def Configure(self, argin):
...
-            argin1 = [float(argin[0]), float(argin[1])]
-            spectrum = [0]
-            spectrum.extend((argin1))
-            self._dish_proxy.desiredPointing = spectrum
-            self._dish_proxy.command_inout_asynch(CONST.CMD_DISH_SLEW, "0", self.commandCallback)
+            # Convert ra and dec to az and el
+            self.convert_radec_to_azel(argin)
+
+            # Invoke slew command on DishMaster with az and el as inputs
+            #argin1 = [float(argin[0]), float(argin[1])]
```
Comment (jyotinpo) [inline tmcprototype/DishLeafNode/DishLeafNode/DishLeafNode.py]: This commented line 468 can be removed.

**Pull request merged**